clouddriver/clouddriver-azure/src/test/java/com/netflix/spinnaker/clouddriver/azure/security/AzureAccountDefinitionSourceTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025 Harness, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.azure.security;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.netflix.spinnaker.clouddriver.azure.config.AzureConfigurationProperties;
+import com.netflix.spinnaker.clouddriver.security.AccountDefinitionRepository;
+import com.netflix.spinnaker.credentials.definition.CredentialsDefinitionSource;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+
+public class AzureAccountDefinitionSourceTest {
+
+  @Test
+  void testAccountDefinitionSourceReturnsAccounts() {
+    // Arrange
+    AzureConfigurationProperties mockProperties = mock(AzureConfigurationProperties.class);
+    AccountDefinitionRepository mockRepository = mock(AccountDefinitionRepository.class);
```

Comment:
Not against this being a mock per se on a repo, but similarly... there may be a better way than mocking this...